charms/worker/k8s/src/charm.py

```diff
@@ -149,10 +151,47 @@ def __init__(self, *args):
         )
 
         self.framework.observe(self.on.update_status, self._on_update_status)
+        self.framework.observe(self.on.feature_relation_changed, self._on_feature_relation_changed)
+
         if self.is_control_plane:
             self.etcd = EtcdReactiveRequires(self)
             self.framework.observe(self.on.get_kubeconfig_action, self._get_external_kubeconfig)
 
+    def _on_feature_relation_changed(self, event: ops.RelationChangedEvent):
+        relation_data = event.relation.data.get(event.unit) #type: ignore
+        if not relation_data:
+            log.warning("No relation data found for unit %s", event.unit)
+            return
+
+        feature_name = relation_data.get("feature-name")
+        feature_version = relation_data.get("feature-version") # library version
+        feature_attributes = relation_data.get("feature-attributes")
+
+        feature_config_classes: Dict[str, type] = {
+            "load-balancer": LoadBalancerConfig,
+            "local-storage": LocalStorageConfig,
+        }
+
+        if feature_name and feature_version and feature_attributes:
+            feature_attributes = json.loads(feature_attributes)
+
+            config_class = feature_config_classes.get(feature_name)
+            if config_class:
+                feature_config = config_class(**feature_attributes)
+
+                feature_name = feature_name.replace("-", "_")
+                config = UserFacingClusterConfig(**{feature_name: feature_config})
+                update_request = UpdateClusterConfigRequest(config=config)
+                self.api_manager.update_cluster_config(update_request)
+
+                log.info("Got feature of type [%s], with version [%s] and attributes [%s]", feature_name, feature_version, json.dumps(feature_config.dict()))
+            else:
+                log.warning("Unsupported feature name: %s", feature_name)
```

Comment:
Here's a good situation to "block the charm" 
you can do this with `status.add(ops.BlockedStatus(f"Unsupported feature {feature_name}"))`